### PR TITLE
Source `total_score` from `legacy_total_score` when present

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] Updated osu.Game with https://github.com/ppy/osu/pull/20028

With guidance from @notbakaneko on how to adjust the sort.

### CODE REVIEW

**Review by peppy — DISMISSED** [review]

File: osu.ElasticIndexer/schemas/solo_scores.json [hunk 1]
```diff
@@ -62,6 +62,9 @@
       },
       "user_id": {
         "type": "keyword"
+      },
+      "is_legacy": {
```
Comment (notbakaneko) [inline osu.ElasticIndexer/schemas/solo_scores.json:66]: should go between `id` and `max_combo`

Comment (smoogipoo) [reply osu.ElasticIndexer/schemas/solo_scores.json]: Done. Is this for style, or does it have a logical effect that I can test somehow?

Comment (notbakaneko) [reply osu.ElasticIndexer/schemas/solo_scores.json]: just style.

**Comment by notbakaneko** [comment]
I saw the purple icon and checked if out of reflex, forgetting that project dependencies are separate 😅

**Review by peppy — APPROVED** [review]

**Pull request merged**